lib/ocpp/common/websocket/websocket_libwebsockets.cpp

```diff
@@ -713,17 +717,23 @@ bool WebsocketLibwebsockets::connect() {
     // use new connection context
     conn_data = local_data;
 
-    // Wait old thread for a clean state
-    if (this->websocket_thread) {
-        // Awake libwebsockets thread to quickly exit
-        request_write();
-        this->websocket_thread->join();
-    }
+    {
+        std::scoped_lock lock(connection_mutex);
+
+        // Wait old thread for a clean state
+        if (this->websocket_thread && this->websocket_thread->joinable()) {
+            // Awake libwebsockets thread to quickly exit
+            request_write();
+            this->websocket_thread->join();
+            this->websocket_thread.reset();
+        }
 
-    if (this->recv_message_thread) {
-        // Awake the receiving message thread to finish
-        recv_message_cv.notify_one();
-        this->recv_message_thread->join();
+        if (this->recv_message_thread && this->websocket_thread->joinable()) {
```

Comment:
```suggestion
        if (this->recv_message_thread && this->recv_message_thread->joinable()) {
```

Reply:
Thanks, good find. Too quick with copying :speak_no_evil:

Reply:
Actually caught this by running this PR with the [tests added here](https://github.com/EVerest/everest-core/pull/959) - seems to be valuable ;)